File: exercises/03 annoying blobs/cpp/AnnoyingBlobs/AnnoyingBlobs/BlobViewer.cpp
#include "DataDisplayer.h"
#include "DataAdder.h"

// gtest_main.lib already introduces a main, so for the sake of the exercise we call this one realMain. Just call it from
// your test.
void realMain()
{
	DataDisplayer view;
	DataAdder model;

	for (int j = 0; j < 30; ++j)
	{
		if ((j % 3) == 0)
		{
			model.doSomeAdding();
		}

		view.tick();
	}
}

File: solutions/03 annoying blobs/java/src/eu/sioux/tdd/blob/BlobViewer.java
package eu.sioux.tdd.blob;

public class BlobViewer {

    /**
     * @param args
     */
    public static void main(String[] args) {
        DataBlob dataBlob = new DataBlobFactory().createDataBlob();
        DataDisplayer view = new DataDisplayer(dataBlob);
        DataAdder model = new DataAdder(dataBlob);

        for (int j = 0; j < 30; j++) {
            if ((j % 3) == 0) {
                model.doSomeAdding();
            }

            view.tick();
        }

    }

}
